modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -15,447 +15,423 @@
  */
 package org.testcontainers.couchbase;
 
-import com.couchbase.client.core.utils.Base64;
-import com.couchbase.client.java.Bucket;
-import com.couchbase.client.java.CouchbaseCluster;
-import com.couchbase.client.java.cluster.*;
-import com.couchbase.client.java.env.CouchbaseEnvironment;
-import com.couchbase.client.java.env.DefaultCouchbaseEnvironment;
-import com.couchbase.client.java.query.Index;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.github.dockerjava.api.command.ExecCreateCmdResponse;
 import com.github.dockerjava.api.command.InspectContainerResponse;
-import com.google.common.collect.Lists;
-import lombok.*;
-import org.apache.commons.compress.utils.Sets;
-import org.apache.commons.io.IOUtils;
-import org.jetbrains.annotations.NotNull;
+import com.github.dockerjava.api.model.ContainerNetwork;
+import okhttp3.Credentials;
+import okhttp3.FormBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
 import org.testcontainers.containers.GenericContainer;
-import org.testcontainers.containers.Network;
-import org.testcontainers.containers.SocatContainer;
 import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
-import org.testcontainers.images.builder.Transferable;
-import org.testcontainers.utility.ThrowingFunction;
+import org.testcontainers.containers.wait.strategy.WaitAllStrategy;
 
-import java.io.DataOutputStream;
 import java.io.IOException;
-import java.net.HttpURLConnection;
-import java.net.URL;
-import java.net.URLEncoder;
-import java.nio.charset.StandardCharsets;
-import java.util.*;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicReference;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
-
-import static java.net.HttpURLConnection.HTTP_OK;
-import static org.testcontainers.couchbase.CouchbaseContainer.CouchbasePort.*;
 
 /**
- * Based on Laurent Doguin version,
+ * The couchbase container initializes and configures a Couchbase Server single node cluster.
  * <p>
- * optimized by Tayeb Chlyah
+ * Note that it does not depend on a specific couchbase SDK, so it can be used with both the Java SDK 2 and 3 as well
+ * as the Scala SDK 1 or newer. We recommend using the latest and greatest SDKs for the best experience.
  */
-@AllArgsConstructor
 public class CouchbaseContainer extends GenericContainer<CouchbaseContainer> {
 
-    public static final String VERSION = "5.5.1";
-    public static final String DOCKER_IMAGE_NAME = "couchbase/server:";
```

Comment:
I know this is a big breaking change after all, but in other modules, we sometimes also have image name and tag public.

Reply:
see #2054 for the discussion